compass/ocean/tests/merry_go_round/initial_state.py

```diff
@@ -0,0 +1,192 @@
+
+        dsMesh = cull(dsMesh, logger=logger)
+        dsMesh = convert(dsMesh, graphInfoFileName='culled_graph.info',
+                         logger=logger)
+        write_netcdf(dsMesh, 'culled_mesh.nc')
+
+        ds = dsMesh.copy()
+        angleEdge = ds.angleEdge.values
+        xCell = ds.xCell
+        yCell = ds.yCell
+        xEdge = ds.xEdge
+        xOffset = numpy.min(xEdge.values)
+        xCellAdjusted = xCell - xOffset
+        xEdgeAdjusted = xEdge - xOffset
+        nCells = ds['nCells'].size
+        nEdges = ds['nEdges'].size
+
+        ds['bottomDepth'] = bottom_depth * xarray.ones_like(xCell)
+        ds['ssh'] = xarray.zeros_like(xCell)
+
+        # Note: machine-precision diffs in layerThickness are present from
+        # legacy compass which did not use init_vertical_coord
+        config.set('vertical_grid', 'vert_levels', str(res_params['nz']))
+        init_vertical_coord(config, ds)
+
+        nVertLevels = ds['nVertLevels'].size
+
+        xMin = xCellAdjusted.min()
+        xMax = xCellAdjusted.max()
+        xMid = 0.5*(xMin + xMax)
+        zMid = ds.zMid.values
+
+        # Only the top layer moves in this test case
+        vertCoordMovementWeights = xarray.ones_like(ds.refZMid)
+        if (vert_coord == 'z-level'):
+            vertCoordMovementWeights[:] = 0.0
+            vertCoordMovementWeights[0] = 1.0
+        ds['vertCoordMovementWeights'] = vertCoordMovementWeights
+
+        # Initialize temperature
+        xCellDepth, _ = xarray.broadcast(xCellAdjusted, ds.refBottomDepth)
+        xEdgeDepth, _ = xarray.broadcast(xEdgeAdjusted, ds.refBottomDepth)
+        angleEdgeDepth, _ = xarray.broadcast(ds.angleEdge, ds.refBottomDepth)
+        xCellAdjusted = xCellAdjusted.values
+        xEdgeAdjusted = xEdgeAdjusted.values
+        temperature = temperature_right * xarray.ones_like(xCellDepth)
+        temperature = xarray.where(xCellDepth < xMid,
+                                   temperature_left,
+                                   temperature_right)
+        temperature = temperature.expand_dims(dim='Time', axis=0)
+        ds['temperature'] = temperature
+
+        # Initialize temperature
+        ds['salinity'] = salinity_background * xarray.ones_like(temperature)
+
+        # Initialize normalVelocity
+        normalVelocity = xarray.zeros_like(xEdgeDepth)
+        zMidEdge = xarray.zeros_like(xEdgeDepth)
+        mask = xarray.zeros_like(xEdgeDepth, dtype='bool')
+        cell1 = ds.cellsOnEdge[:, 0].values - 1
+        cell2 = ds.cellsOnEdge[:, 1].values - 1
+        for iEdge in range(0, nEdges):
+            if ((cell1[iEdge] < 0.5*nCells and
+                 cell1[iEdge] > 0.25*nCells - 1) and
+                (cell2[iEdge] < 0.5*nCells and
+                 cell2[iEdge] > 0.25*nCells - 1)):
+                mask[iEdge] = True
+            elif (cell1[iEdge] > 0.75*nCells-1 and
+                  cell2[iEdge] > 0.75*nCells-1):
+                mask[iEdge] = True
```

Comment:
This would be substantially more efficient if it used `numpy.logical_and` and `numpy.logical_or`.  It would also seems strange to use the cell index rather than a physical coordinate _(I think `xCell` in this case) to do this masking.  As it is, I don't entirely understand what this code is doing.

```suggestion
x1 = xCell[cell1]
x2 = xCell[cell2]
xQuarter = 0.75*xMin + 0.25*xMax
xThreeQuarters = 0.25*xMin + 0.75*xMax
mask = numpy.logical_or(
    numpy.logical_and(
        numpy.logical_and(x1 < xMid, x1 >= xQuarter),
        numpy.logical_and(x2 < xMid, x2 >= xQuarter)),
    numpy.logical_and(x1 > xThreeQuarters, 
                      x2 > xThreeQuarters)))
```